openapi_core/schema/schemas/models.py

```diff
@@ -216,7 +220,25 @@ def _unmarshal_string(self, value, custom_formatters=None):
             raise InvalidCustomFormatSchemaValue(
                 "Failed to format value {value} to format {type}: {exception}", value, self.format, exc)
 
-    def _unmarshal_any(self, value, custom_formatters=None):
+    def _unmarshal_integer(self, value, custom_formatters=None, strict=True):
+        if strict and not isinstance(value, (integer_types, )):
+            raise InvalidSchemaValue("Value {value} is not of type {type}", value, self.type)
+
+        return int(value)
+
+    def _unmarshal_number(self, value, custom_formatters=None, strict=True):
+        if strict and not isinstance(value, (float, )):
```

Comment:
I have a case where I return a number with a value of 3
3 contrary to 3.0 is not recognized as a float.